library/std/src/os/unix/net/listener.rs

```diff
@@ -73,9 +73,13 @@ impl UnixListener {
         unsafe {
             let inner = Socket::new_raw(libc::AF_UNIX, libc::SOCK_STREAM)?;
             let (addr, len) = sockaddr_un(path.as_ref())?;
+            #[cfg(target_os = "linux")]
+            const backlog: libc::c_int = -1;
```

Comment:
Does this actually work? The posix documentation says

```
If listen() is called with a backlog argument value that is less than 0, the function behaves as if it had been called with a backlog argument value of 0.

A backlog argument of 0 may allow the socket to accept connections, in which case the length of the listen queue may be set to an implementation-defined minimum value. 
```

Note that's the implementation-defined *minimum value*.

The linux-specific documentation is silent on negative numbers and only covers the *maximum value*

       If the backlog argument is greater than the value in
       /proc/sys/net/core/somaxconn, then it is silently capped to that
       value.  Since Linux 5.4, the default in this file is 4096; in
       earlier kernels, the default value is 128.  In kernels before
       2.4.25, this limit was a hard coded value, SOMAXCONN, with the
       value 128.

Reply:
yes it works on Linux (in fact it works on few more oses too (e.g. macOs, FreeBSD,...)) as it falls to kernel default (so here 128 or 4096) or the setting you mentioned earlier.

Reply:
I would like to avoid using undocumented behavior. Sure, linus says "don't break userspace" but that doesn't mean we should rely on every implementation detail we can find. And the BSDs might have other policies

Is there any documentation, mailing list entry, commit message or whatever that tells us that -1 is intended to mean "default". Only freebsd documents that, and even there it's only in the history section, not in the function parameter description 🙄.

Reply:
indeed not much of documentation, however here the source code of FreeBSD handling negative backlog
https://github.com/freebsd/freebsd-src/blob/e7d02be19d40063783d6b8f1ff2bc4c7170fd434/sys/kern/uipc_socket.c#L1069

here a reference for Linux
https://github.com/torvalds/linux/commit/19f92a030ca6d772ab44b22ee6a01378a8cb32d4

Reply:
Ok, if the kernel devs are aware and not complaining about how userspace is using it then it's fine I guess. Still not great that it's so underdocumented.